dataset/sort.cpp

```diff
@@ -27,7 +16,7 @@ Variable sort(const Variable &var, const Variable &key, const SortOrder order) {
 DataArray sort(const DataArray &array, const Variable &key,
                const SortOrder order) {
   auto helper = array;
-  const auto dummy = nonclashing_name(helper.coords());
+  const Dim dummy = Dim::Internal0;
```

Comment:
This does not work if the caller of `sort` already inserted `Dim::Internal0` for some reason.

Reply:
If this turns out to be a problem we may need to switch to function-specific internal flags. At this point I think this is unlikely to happen.

Reply:
But it would be painful to figure out once it does happen.

Reply:
Changed to dedicated dim IDs